File: Dsa-day230.cpp
class Solution {
public:
    int minOperations(string s) {
        int n=s.size();
        string s1="";
        string s2="";
        int cnt1=0;
        int cnt2=0;
        for(int i=0;i<n;i++){
            if(i%2==0) {
                s1+="1";
                s2+="0";
            }
            else {
                s1+="0";
                s2+="1";
            }
        }
        for(int i=0;i<n;i++){
            if(s[i]!=s1[i]) cnt1++;
            if(s[i]!=s2[i]) cnt2++;
        }
            
        return min(cnt1,cnt2);
    }
};

File: Dsa-day230.java
class Solution {
    public int minOperations(String s) {
       int c0 = 0;
       int c1 = 0;
       boolean flag = true;

       for(char ch : s.toCharArray()){
            if(flag && ch == '1') c0++;
            else if(!flag && ch == '0') c0++;

            if(flag && ch == '0') c1++;
            else if(!flag && ch == '1') c1++;

            flag = !flag;
       }

       return Math.min(c0, c1);
    }
}